third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -1063,12 +1059,8 @@ fn check_depth_of_type_impl(
                     check_depth_of_type_impl(resolver, ty, max_depth, check_depth!(0))
                 })
                 .collect::<PartialVMResult<Vec<_>>>()?;
-            let struct_type = resolver.loader().get_struct_type(*si)?;
-            check_depth!(struct_type
-                .depth
-                .as_ref()
-                .ok_or_else(|| { PartialVMError::new(StatusCode::VM_MAX_VALUE_DEPTH_REACHED) })?
-                .solve(&ty_arg_depths))
+            let formula = resolver.loader().calculate_depth_of_struct(*si)?;
```

Comment:
The motivation is to get ready for module invalidation. Depth formula is a property that could only be calculated by fetching all transitive deps. With this PR I'd like to make module resolution to be not dependent on the deps so that's why I moved it.